pkg/model/crd.go

```diff
@@ -2173,23 +2202,24 @@ func (r *CRD) goCodeSetOutputForContainer(
 			//		f0elem := string{}
 			out += r.goCodeVarEmptyConstructorK8sType(
 				valVarName,
-				shape.ValueRef.Shape,
+				targetShape.ValueRef.Shape,
```

Comment:
Because targetShape is OutputShape of CreateInput(spec) or CreateOutput(status) so we should be creating constructors based on targetShape.

--------------
The issue was surfaced during creation of APIGatewayV2 Integration sdk.go file
Integration has a field called "TlsConfig".
The outputShape of ReadOne has the name "TlsConfig", while the inputShape of Create object (spec) has the field called "TlsConfigInput"

If we use the sourceShape to create constructor, i.e. TlsConfig, we get compile time errors that TlsConfig does not exist in ko.Spec

----------------

My approach here is to
a) Introduce targetShape alongside sourceShape in every method. TargetShape is either spec or status shape. while sourceshape can also be readOne or readMany shape.
b) If any member of sourceShape member does not exist in targetShape, ignore it.
c) Perform constructor creation based on targetShape and not source Shape.